numba/typeinfer.py

```diff
@@ -256,6 +259,9 @@ def __call__(self, typeinfer):
                     tup = types.UniTuple(dtype=tp.iterator_type.yield_type,
                                          count=self.count)
                     typeinfer.add_type(self.target, tup, loc=self.loc)
+                    break
+            else:
```

Comment:
Yes, this branch provide better information on the error.  We should make sure it stays there.  Originally, it just reports "Undefined variable" on such error.